libcxxabi/test/test_memory_alloc.pass.cpp

```diff
@@ -0,0 +1,34 @@
+//===----------------------------------------------------------------------===//
```

Comment:
We have a bunch of tests located in e.g. `libcxx/test/std/language.support/support.dynamic/new.delete/new.delete.array/new.size.pass.cpp`. Those are testing the various version of `operator new` we have:

```c++
void* operator new(std::size_t);
void* operator new[](std::size_t);
void* operator new(std::size_t, std::align_val_t);
void* operator new[](std::size_t, std::align_val_t);
```

You'll see each overload has its own test file. Can you add one test for each such overload? Note that we also have `std::nothrow_t` overloads of `operator new` but I don't think it makes sense to test that the `new_handler` throws `bad_alloc` for those, since the only thing it would do is terminate.

Reply:
> You'll see each overload has its own test file. Can you add one test for each such overload? Note that we also have `std::nothrow_t` overloads of `operator new` but I don't think it makes sense to test that the `new_handler` throws `bad_alloc` for those, since the only thing it would do is terminate.

@ldionne @azhan92: If I read the following correctly, testing the `std::nothrow_t` overloads is meaningful (to ensure that the exception is caught by the default implementation):

> 
> Default behavior: Calls operator new(size), or operator new(size, alignment), respectively. If
> the call returns normally, returns the result of that call. Otherwise, returns a null pointer.

Reply:
Ah, yes, you're right!